src/components/navigation-bar/navigation-bar.test.js

```diff
@@ -0,0 +1,169 @@
+const specialCharacters = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
+const testData = "cypressData";
+const variants = [
+  ["black", "rgb(0, 0, 0)"],
+  ["light", "rgb(230, 235, 237)"],
+  ["white", "rgb(255, 255, 255)"],
+  ["dark", "rgb(0, 50, 76)"],
+];
+
+const NavigationBarComponent = ({ children, ...props }) => {
+  const isChildren = children != null ? children : "Cypress tests";
+  return <NavigationBar {...props}>{isChildren}</NavigationBar>;
+};
+
+context("Testing NavigationBar component", () => {
+  describe("should render NavigationBar component", () => {
+    it.each(specialCharacters)(
+      "should render NavigationBar component with %s as a children",
+      (childrenValue) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent>{childrenValue}</NavigationBarComponent>
+        );
+
+        navigationBar().should("have.text", childrenValue);
+      }
+    );
+
+    it("should render NavigationBar component with ariaLabel prop", () => {
+      CypressMountWithProviders(
+        <NavigationBarComponent ariaLabel={testData} />
+      );
+
+      navigationBar().should("have.attr", "aria-label", testData);
+    });
+
+    it.each(variants)(
+      "should render NavigationBar component with %s as a navigationType",
+      (navigationType, color) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent navigationType={navigationType} />
+        );
+
+        navigationBar().should("have.css", "background-color", color);
+      }
+    );
+
+    it.each([
+      [true, "not.exist"],
+      [false, "be.visible"],
+    ])(
+      "should render NavigationBar component with isLoading prop set to %s",
+      (boolean, assertion) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent isLoading={boolean}>
+            {testData}
+          </NavigationBarComponent>
+        );
+
+        cy.contains(testData).should(assertion);
+      }
+    );
+
+    it.each([["fixed"], ["sticky"]])(
+      "should render NavigationBar component with position prop set to %s with orientation set to top",
+      (position) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent position={position} orientation="top" />
+        );
+
+        navigationBar().should("have.css", "position", position);
+      }
+    );
+
+    it.each([["fixed"], ["sticky"]])(
+      "should render NavigationBar component with position prop set to %s with orientation set to bottom",
+      (position) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent position={position} orientation="bottom" />
+        );
+
+        navigationBar().should("have.css", "position", position);
+      }
+    );
+
+    it.each([[25], [100], [-100]])(
+      "should render NavigationBar component with offset prop set to %s",
+      (offset) => {
+        CypressMountWithProviders(<NavigationBarComponent offset={offset} />);
+
+        navigationBar().should("have.attr", "offset", offset);
```

Comment:
I don't think this is the right way to test the `offset` prop. This only works because the rendered output does include `offset="25px"` as an attribute - I'm not actually sure what this does but I think it's a unintended consequence of spreading `...rest` onto the underlying element.

What this prop is actually doing is providing the value of the CSS `top` or `bottom` property - which one depends on the `orientation`. (See line 61 of `navigation-bar.style.ts`.) So that CSS is what we need to test.

Reply:
why? we are providing the `offset` to some values and then just asserting that those values were rendered properly. Like in screenshot below. 
<img width="505" alt="Screenshot 2022-10-10 at 07 50 34" src="https://user-images.githubusercontent.com/34001198/194805216-4a6826cc-26db-4831-a3af-5a9250272841.png">

Reply:
hi Igor, I must have explained it badly :-)

What I meant was that, while the `offset` prop is indeed setting an `offset` attribute on the HTML element, which is what you're testing, as far as I can tell this attribute is an accident and doesn't actually do anything. The important thing about what this prop does is that it sets the CSS `top` or `bottom` value (which one depends on the `orientation` prop) to the value of this prop. So I think that's what we should be testing.

By all means ask James if you want a second opinion, as he's the one who implemented this component so will probably have more understanding than me. I could be wrong but I expect he'll be surprised like I was to find `offset` and `orientation` as actual attributes in the DOM. (They are passed through by styled components automatically, presumably because they're technically valid attributes - although I couldn't find them documented anywhere...)